File: src/main/java/cal/fstg/java/service/ResumeService.java
//package cal.fstg.java.service;
//
//import cal.fstg.java.dao.ResumeRepository;
//import cal.fstg.java.dao.StudentRepository;
//import cal.fstg.java.model.Resume;
//import cal.fstg.java.model.Student;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.stereotype.Service;
//
//import java.util.List;
//import java.util.Optional;
//
//@Service
//public class ResumeService {
//
//    private final ResumeRepository resumeRepository;
//
//    @Autowired
//    public ResumeService(ResumeRepository resumeRepository) {
//        this.resumeRepository = resumeRepository;
//    }
//
//    public List<Resume> getAll() {
//        return resumeRepository.findAll();
//    }
//
//    public Optional<Resume> getOne(long id) {
//        return resumeRepository.findById(id);
//    }
//
//    public Resume save(Resume resume) {
//        return resumeRepository.save(resume);
//    }
//
//    public void delete(long id) {
//        resumeRepository.deleteById(id);
//    }
//}

File: src/main/kotlin/cal/fstg/stagerkotlin/service/ResumeService.kt
package cal.fstg.stagerkotlin.service

import cal.fstg.stagerkotlin.dao.ResumeRepository
import cal.fstg.stagerkotlin.model.Resume
import cal.fstg.stagerkotlin.model.User
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.reactor.flux
import kotlinx.coroutines.reactor.mono
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ResumeService(@Autowired private val resumeRepository: ResumeRepository) {

    @ExperimentalCoroutinesApi
    fun getAll() = flux<Resume> { resumeRepository.findAll() }

    fun getOne(id: String) = mono { resumeRepository.findById(id) }

    fun save(resume: Resume) = mono { resumeRepository.save(resume) }

    fun delete(id: String) = resumeRepository.deleteById(id)
}